### ISSUE TITLE
[BUG] intersight_networkconfig_policy IPv6 Issue

### ISSUE THREAD

**Issue by scotttyso** [CiscoDevNet/terraform-provider-intersight#104]
**Describe the bug**
with the resource "intersight_networkconfig_policy".  If IPv6 is disabled a repeated terraform apply will continually return a change for the DNS IPv6 servers with a value of "::"

Example Here

```

  # module.dns_example_1.intersight_networkconfig_policy.dns will be updated in-place
  ~ resource "intersight_networkconfig_policy" "dns" {
      - alternate_ipv6dns_server = "::" -> null
        id                       = "60d727bd6275722d3182bca1"
        name                     = "example_1"
      - preferred_ipv6dns_server = "::" -> null
      - 
```

But with IPv6 disabled the provider will not allow me to push "::" as part of the initial policy creation.  If I try to apply "::" as part of the initial policy push I get the following error:

```
╷
│ Error: error occurred while creating NetworkconfigPolicy: 400 Bad Request Response from endpoint: {"code":"InvalidRequest","message":"Cannot create policy with IPv6 DNS Server as policy is not configured with IPv6 enabled.","messageId":"cannot_create_policy_with_ipv6_dns_ipv6_disable","messageParams":{},"traceId":"NBa197e2d9307255b539692dbbe99acb48"}
│ 
│   with module.dns_example_1.intersight_networkconfig_policy.dns,
│   on ../../modules/policies_network_connectivity/main.tf line 14, in resource "intersight_networkconfig_policy" "dns":
│   14: resource "intersight_networkconfig_policy" "dns" {
│ 
```

So it is a catch 22.  I can work around it by applying the policy and then modifying the policy after the push to define "::" but this is not the desired effect of creating these modules for users.

Below is my example configuration.

```
resource "intersight_networkconfig_policy" "dns" {
  alternate_ipv4dns_server  = length(var.dns_servers_v4) > 1 ? var.dns_servers_v4[1] : null
  alternate_ipv6dns_server  = length(var.dns_servers_v6) > 1 ? var.dns_servers_v6[1] : null
  description               = var.description
  dynamic_dns_domain        = var.update_domain
  enable_dynamic_dns        = var.dynamic_dns
  enable_ipv4dns_from_dhcp  = var.dynamic_dns == true ? true : false
  enable_ipv6               = var.ipv6_enable
  enable_ipv6dns_from_dhcp  = var.ipv6_enable == true && var.dynamic_dns == true ? true : false
  preferred_ipv4dns_server  = length(var.dns_servers_v4) > 0 ? var.dns_servers_v4[0] : null
  preferred_ipv6dns_server  = length(var.dns_servers_v6) > 0 ? var.dns_servers_v6[0] : null
  name                      = var.name
  organization {
    moid        = var.org_moid
    object_type = "organization.Organization"
  }
  dynamic "profiles" {
    for_each = var.profiles
    content {
      moid        = profiles.value.moid
      object_type = profiles.value.object_type
    }
  }
  dynamic "tags" {
    for_each = var.tags
    content {
      key   = tags.value.key
      value = tags.value.value
    }
  }
}
```

**To Reproduce**
1. Resource for '...'
2. Run `terraform apply` '....'
3. Update resource '....'

**Expected behavior**
We either need to allow a user to define "::" for the IPv6 DNS server even when IPV6 is disabled.  Or when  IPv6 is disabled, the ipv6 dns attributes should be null not "::".

**Screenshots and Terraform configurations**

**Provider Version :** v1.0.11

**Additional context**

**Comment by gve-vse-tim** [comment]
This issue has existed since the provider has been released as I documented in #15 

In my mind, the quick workaround would be that the Terraform provider should be coded in such a way that TF's "null" is equivalent to Intersight's "::" for these two attributes (the preferred and alternate ipv6 DNS servers).

Given Intersight's backend behavior (setting a default value for IPv6 attributes when IPv6 is disabled), I can't foresee a scenario where that equivalence would behave badly.

Although, honestly, the correct fix would be to stop Intersight from populating those defaults when IPv6 is disabled.

**Comment by jerewill-cisco** [comment]
Calling this an "enhancement" and not a "bug" is not good.  It has allowed this bug to hang around for far too long.  The resource drift detection is not working correctly and I think that I agree with @gve-vse-tim.  A SchemaDiffSuppressFunc could be used on these two fields to suppress the drift easily.

**Comment by meghana-rajashekar** [comment]
#### The issue has been resolved in the latest provider release(1.0.23) to explicitly ignore the ipv6 attributes when ipv6_enabled is set to false.
- [x] Expected behavior: There should be no diffs on the subsequent apply
